crates/bevy_ui/src/render/ui.wgsl

```diff
@@ -30,6 +33,10 @@ var sprite_sampler: sampler;
 @fragment
 fn fragment(in: VertexOutput) -> @location(0) vec4<f32> {
     var color = textureSample(sprite_texture, sprite_sampler, in.uv);
```

Comment:
Perhaps move the texture sample into the branch?

Reply:
This is a limitation of fragment shaders. For maximum compatibility you have to sample the texture on all code paths, regardless of whether you use the result. I don't think it will even compile the other way.

I'll add a comment to the shader.